### PR TITLE
You're the computer, you sort the fields

### PR DESCRIPTION
**Pull request by aboodman**
Restructure makeStructType to have sane API, update callers.

### CODE REVIEW

**Comment by aboodman** [comment]
cc: @ahl

File: js/noms/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@attic/noms",
   "license": "Apache-2.0",
-  "version": "60.0.1",
+  "version": "61.0.1",
```
Comment (arv) [inline js/noms/package.json]: 61.0.0

```
npm version major
```

next time

File: js/noms/src/type.js [hunk 2]
```diff
@@ -195,9 +195,9 @@ export function makeRefType(elemType: Type<any>): Type<CompoundDesc> {
   return staticTypeCache.getCompoundType(Kind.Ref, elemType);
 }
 
-export function makeStructType(name: string, fieldNames: string[], fieldTypes: Type<any>[]):
+export function makeStructType(name: string, fields: { [key: string]: Type<any> }) :
```
Comment (arv) [inline js/noms/src/type.js]: no ws after `{`... I guess eslint is still not checkin flow types.

**Comment by arv** [comment]
So much nicer! 😎

**Pull request merged**